### ISSUE TITLE
vscode-extensions.vadimcn.vscode-lldb: Error `failed to get reply to handshake packet` on `x86_64-darwin`

### ISSUE THREAD

**Issue by nigelgbanks** [NixOS/nixpkgs#148946]
### Describe the bug

On `x86_64-darwin` the VS Code plugin `vscode-extensions.vadimcn.vscode-lldb` fails to connect to the debugger after starting. With the message `failed to get reply to handshake packet`.

If I install VSCode **and** the plugin manually it works as expected. That being said others **not** using `nixpkgs` have encountered [this issue](https://github.com/vadimcn/vscode-lldb/issues/323) as well.  I've put some [effort](https://github.com/vadimcn/vscode-lldb/issues/323#issuecomment-985565648) into debugging the issue though not much luck.

### Steps To Reproduce

I've attached a minimal reproducible example project that uses flakes.

[Issue.zip](https://github.com/NixOS/nixpkgs/files/7659537/Issue.zip)

1. On `x86_64-darwin` machine.
2. Ensure VS Code is uninstalled
     - Remove the application via Nix if installed by Nix or manually from the `/Applications` folder.
     - `rm -fr $HOME/Library/Application\ Support/Code/` 
     - `rm -rf $HOME/.vscode`
2. Change into the directory of the unpacked zip file.
3. Run `nix develop`.
4. Once in the shell use `code .` to open visual studio code.
5. Put a breakpoint in `src/main.rc`
6. Launch the debugger.
7. Observe the error.

### Expected behavior

The debugger attaches to the process and stops on the set breakpoint.

### Screenshots

<img width="1022" alt="Screenshot 2021-12-06 at 08 33 20" src="https://user-images.githubusercontent.com/487373/144813378-e73ae4b0-cbd1-458f-b93d-58ea0373af21.png">

The following is the log output of `codelldb`.

```
Running `cargo build --bin=youtube-dl-subscriptions --package=youtube-dl-subscriptions --message-format=json`...
    Finished dev [unoptimized + debuginfo] target(s) in 1.33s
Raw artifacts:
{
  fileName: '/Users/nbanks/Projects/youtube-dl-subscriptions/target/debug/youtube-dl-subscriptions',
  name: 'youtube-dl-subscriptions',
  kind: 'bin'
}
Filtered artifacts: 
{
  fileName: '/Users/nbanks/Projects/youtube-dl-subscriptions/target/debug/youtube-dl-subscriptions',
  name: 'youtube-dl-subscriptions',
  kind: 'bin'
}
configuration: {
  type: 'lldb',
  request: 'launch',
  name: "Debug executable 'youtube-dl-subscriptions'",
  args: [],
  cwd: '${workspaceFolder}',
  __configurationTarget: 5,
  relativePathBase: '/Users/nbanks/Projects/youtube-dl-subscriptions',
  program: '/Users/nbanks/Projects/youtube-dl-subscriptions/target/debug/youtube-dl-subscriptions',
  sourceLanguages: [ 'rust' ]
}
liblldb: /Users/nbanks/.vscode/extensions/vadimcn.vscode-lldb/lldb/lib/liblldb.dylib
environment: {}
params: {
  evaluateForHovers: true,
  commandCompletions: true,
  sourceLanguages: [ 'rust' ]
}
Listening on port 56932
[2021-12-06T08:27:59.692Z DEBUG codelldb] New debug session
INFO(Python) 08:28:00 formatters: Initializing
INFO(Python) 08:28:00 formatters.rust: Initializing
[2021-12-06T08:28:00.526Z DEBUG codelldb::dap_codec] --> {"command":"initialize","arguments":{"clientID":"vscode","clientName":"Visual Studio Code","adapterID":"lldb","pathFormat":"path","linesStartAt1":true,"columnsStartAt1":true,"supportsVariableType":true,"supportsVariablePaging":true,"supportsRunInTerminalRequest":true,"locale":"en-gb","supportsProgressReporting":true,"supportsInvalidatedEvent":true,"supportsMemoryReferences":true},"type":"request","seq":1}
[2021-12-06T08:28:00.550Z DEBUG codelldb::dap_codec] <-- {"type":"response","request_seq":1,"success":true,"command":"initialize","body":{"exceptionBreakpointFilters":[{"default":true,"filter":"rust_panic","label":"Rust: on panic"}],"supportTerminateDebuggee":true,"supportsCancelRequest":true,"supportsCompletionsRequest":true,"supportsConditionalBreakpoints":true,"supportsConfigurationDoneRequest":true,"supportsDataBreakpoints":true,"supportsDelayedStackTraceLoading":true,"supportsEvaluateForHovers":true,"supportsFunctionBreakpoints":true,"supportsGotoTargetsRequest":true,"supportsHitConditionalBreakpoints":true,"supportsLogPoints":true,"supportsReadMemoryRequest":true,"supportsRestartFrame":true,"supportsSetVariable":true}}
[2021-12-06T08:28:00.567Z DEBUG codelldb::dap_codec] --> {"command":"launch","arguments":{"type":"lldb","request":"launch","name":"Debug executable 'youtube-dl-subscriptions'","args":[],"cwd":"/Users/nbanks/Projects/youtube-dl-subscriptions","__configurationTarget":5,"relativePathBase":"/Users/nbanks/Projects/youtube-dl-subscriptions","program":"/Users/nbanks/Projects/youtube-dl-subscriptions/target/debug/youtube-dl-subscriptions","sourceLanguages":["rust"],"_adapterSettings":{"displayFormat":"auto","showDisassembly":"auto","dereferencePointers":true,"suppressMissingSourceFiles":true,"evaluationTimeout":5,"consoleMode":"commands","sourceLanguages":null,"terminalPromptClear":null,"evaluateForHovers":true,"commandCompletions":true,"reproducer":false},"__sessionId":"c9e5ae41-7ea5-4911-a380-59571dfd7dd7"},"type":"request","seq":2}
[2021-12-06T08:28:00.568Z DEBUG codelldb::dap_codec] <-- {"type":"event","seq":1,"event":"capabilities","body":{"capabilities":{"exceptionBreakpointFilters":[{"default":true,"filter":"rust_panic","label":"Rust: on panic"}],"supportTerminateDebuggee":true,"supportsCancelRequest":true,"supportsCompletionsRequest":true,"supportsConditionalBreakpoints":true,"supportsConfigurationDoneRequest":true,"supportsDataBreakpoints":true,"supportsDelayedStackTraceLoading":true,"supportsEvaluateForHovers":true,"supportsFunctionBreakpoints":true,"supportsGotoTargetsRequest":true,"supportsHitConditionalBreakpoints":true,"supportsLogPoints":true,"supportsReadMemoryRequest":true,"supportsRestartFrame":true,"supportsSetVariable":true}}}
[2021-12-06T08:28:01.621Z DEBUG codelldb::dap_codec] <-- {"type":"event","seq":2,"event":"initialized"}
[2021-12-06T08:28:01.622Z DEBUG codelldb::debug_session] Debug event: 0x6000022f8358 Event: broadcaster = 0x7fc78c13c420 (lldb.target), type = 0x00000002 (modules-loaded), data = {youtube-dl-subscriptions}
[2021-12-06T08:28:01.626Z DEBUG codelldb::dap_codec] --> {"command":"setFunctionBreakpoints","arguments":{"breakpoints":[]},"type":"request","seq":3}
[2021-12-06T08:28:01.626Z DEBUG codelldb::dap_codec] <-- {"type":"event","seq":3,"event":"module","body":{"module":{"addressRange":"FFFFFFFFFFFFFFFF","id":"FFFFFFFFFFFFFFFF","name":"youtube-dl-subscriptions","path":"/Users/nbanks/Projects/youtube-dl-subscriptions/target/debug/youtube-dl-subscriptions","symbolFilePath":"/Users/nbanks/Projects/youtube-dl-subscriptions/target/debug/youtube-dl-subscriptions","symbolStatus":"Symbols loaded."},"reason":"new"}}
[2021-12-06T08:28:01.627Z DEBUG codelldb::dap_codec] <-- {"type":"response","request_seq":3,"success":true,"command":"setFunctionBreakpoints","body":{"breakpoints":[]}}
[2021-12-06T08:28:01.628Z DEBUG codelldb::dap_codec] <-- {"type":"request","seq":4,"command":"runInTerminal","arguments":{"args":["/nix/store/84zay225zfzqv4gl1dabhwp72775ff70-vscode-extension-vadimcn-vscode-lldb-1.6.8/share/vscode/extensions/vadimcn.vscode-lldb/adapter/.codelldb-wrapped","terminal-agent","--port=56934"],"cwd":"","kind":"integrated","title":"Debug executable 'youtube-dl-subscriptions'"}}
[2021-12-06T08:28:01.635Z DEBUG codelldb::dap_codec] --> {"command":"setDataBreakpoints","arguments":{"breakpoints":[]},"type":"request","seq":4}
[2021-12-06T08:28:01.636Z DEBUG codelldb::dap_codec] <-- {"type":"response","request_seq":4,"success":true,"command":"setDataBreakpoints","body":{"breakpoints":[]}}
[2021-12-06T08:28:01.780Z DEBUG codelldb::dap_codec] --> {"command":"setExceptionBreakpoints","arguments":{"filters":["rust_panic"]},"type":"request","seq":5}
[2021-12-06T08:28:03.982Z DEBUG codelldb::dap_codec] --> {"type":"response","seq":6,"command":"runInTerminal","request_seq":4,"success":true,"body":{"shellProcessId":11713}}
[2021-12-06T08:28:04.418Z DEBUG codelldb::debug_session] Debug event: 0x6000022e9258 Event: broadcaster = 0x7fc78c13c420 (lldb.target), type = 0x00000001 (breakpoint-changed), data = {}
[2021-12-06T08:28:04.418Z DEBUG codelldb::dap_codec] <-- {"type":"response","request_seq":5,"success":true,"command":"setExceptionBreakpoints"}
[2021-12-06T08:28:04.421Z DEBUG codelldb::dap_codec] --> {"command":"configurationDone","type":"request","seq":7}
[2021-12-06T08:28:04.424Z DEBUG codelldb::dap_codec] <-- {"type":"event","seq":5,"event":"output","body":{"output":"Launching: /Users/nbanks/Projects/youtube-dl-subscriptions/target/debug/youtube-dl-subscriptions\n"}}
[2021-12-06T08:28:05.116Z ERROR codelldb::debug_session] failed to get reply to handshake packet
[2021-12-06T08:28:05.116Z DEBUG codelldb::dap_codec] <-- {"type":"response","request_seq":2,"success":false,"command":"","message":"failed to get reply to handshake packet","show_user":true}
[2021-12-06T08:28:05.116Z DEBUG codelldb::dap_codec] <-- {"type":"response","request_seq":7,"success":true,"command":"configurationDone"}
[2021-12-06T08:28:05.302Z DEBUG codelldb::dap_codec] --> {"command":"disconnect","arguments":{"restart":false},"type":"request","seq":8}
[2021-12-06T08:28:05.313Z DEBUG codelldb::dap_codec] <-- {"type":"response","request_seq":8,"success":true,"command":"disconnect"}
[2021-12-06T08:28:05.627Z DEBUG codelldb::dap_session] Client has disconnected
[2021-12-06T08:28:05.634Z DEBUG codelldb::debug_session] End of the requests stream
[2021-12-06T08:28:05.634Z DEBUG codelldb::debug_session] DebugSession::drop()
[2021-12-06T08:28:05.636Z DEBUG codelldb] Session has ended
[2021-12-06T08:28:05.647Z DEBUG codelldb] Exiting
Debug adapter exit code=0, signal=null.
```

### Additional context

Attempts to resolve the issues and others who are experiencing it in different context is [here](https://github.com/vadimcn/vscode-lldb/issues/323).

### Notify maintainers

@oxalica

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 21.1.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"nixpkgs-22.05pre334535.9f25a8ac3a9"`
 - nixpkgs: `/etc/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
   - vscode-extensions.vadimcn.vscode-lldb
```

**Comment by oxalica** [comment]
Sorry, i no longer use vscode.

**Comment by nigelgbanks** [comment]
@oxalica Fair enough, I'll pick away at this in my spare moments and if I sort this out I'll raise a pull request.

**Comment by arapov** [comment]
this issue is still present.

```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 22.1.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
 ```

```
Initial debug configuration: {
  type: 'lldb',
  request: 'launch',
  name: 'Debug',
  cargo: {
    args: [ 'build', '--bin=play00', '--package=play00' ],
    filter: { name: 'play00', kind: 'bin' }
  },
  args: [],
  cwd: '${workspaceFolder}',
  __configurationTarget: 6
}
Raw artifacts:
{
  fileName: '/Users/anton/Workspace/tutorials/play00/target/debug/play00',
  name: 'play00',
  kind: 'bin'
}
Filtered artifacts: 
{
  fileName: '/Users/anton/Workspace/tutorials/play00/target/debug/play00',
  name: 'play00',
  kind: 'bin'
}
Resolved debug configuration: {
  type: 'lldb',
  request: 'launch',
  name: 'Debug',
  args: [],
  cwd: '${workspaceFolder}',
  __configurationTarget: 6,
  relativePathBase: '/Users/anton/Workspace/tutorials/play00',
  program: '/Users/anton/Workspace/tutorials/play00/target/debug/play00',
  sourceLanguages: [ 'rust' ],
  _adapterSettings: {
    displayFormat: 'auto',
    showDisassembly: 'auto',
    dereferencePointers: true,
    suppressMissingSourceFiles: true,
    evaluationTimeout: 5,
    consoleMode: 'commands',
    sourceLanguages: null,
    terminalPromptClear: null,
    evaluateForHovers: true,
    commandCompletions: true,
    reproducer: false
  }
}
liblldb: /Users/anton/.vscode-oss/extensions/vadimcn.vscode-lldb/lldb/lib/liblldb.dylib
environment: {}
params: {
  evaluateForHovers: true,
  commandCompletions: true,
  sourceLanguages: [ 'rust' ]
}
[2022-12-07T14:33:21.090Z DEBUG codelldb] New debug session
[2022-12-07T14:33:21.148Z ERROR codelldb::debug_session] Initialize Python interpreter: Error:  Fail
    Error Message:
    error: module importing failed: invalid pathname
    
[2022-12-07T14:33:21.148Z DEBUG codelldb::dap_codec] --> {"command":"initialize","arguments":{"clientID":"vscode","clientName":"VSCodium","adapterID":"lldb","pathFormat":"path","linesStartAt1":true,"columnsStartAt1":true,"supportsVariableType":true,"supportsVariablePaging":true,"supportsRunInTerminalRequest":true,"locale":"en-us","supportsProgressReporting":true,"supportsInvalidatedEvent":true,"supportsMemoryReferences":true,"supportsArgsCanBeInterpretedByShell":true},"type":"request","seq":1}
[2022-12-07T14:33:21.148Z DEBUG codelldb::dap_codec] <-- {"seq":1,"type":"response","request_seq":1,"success":true,"command":"initialize","body":{"exceptionBreakpointFilters":[{"default":true,"filter":"rust_panic","label":"Rust: on panic"}],"supportTerminateDebuggee":true,"supportsCancelRequest":true,"supportsCompletionsRequest":true,"supportsConditionalBreakpoints":true,"supportsConfigurationDoneRequest":true,"supportsDataBreakpoints":true,"supportsDelayedStackTraceLoading":true,"supportsEvaluateForHovers":true,"supportsFunctionBreakpoints":true,"supportsGotoTargetsRequest":true,"supportsHitConditionalBreakpoints":true,"supportsLogPoints":true,"supportsReadMemoryRequest":true,"supportsRestartFrame":true,"supportsSetVariable":true,"supportsWriteMemoryRequest":true}}
[2022-12-07T14:33:21.156Z DEBUG codelldb::dap_codec] --> {"command":"launch","arguments":{"type":"lldb","request":"launch","name":"Debug","args":[],"cwd":"/Users/anton/Workspace/tutorials/play00","__configurationTarget":6,"relativePathBase":"/Users/anton/Workspace/tutorials/play00","program":"/Users/anton/Workspace/tutorials/play00/target/debug/play00","sourceLanguages":["rust"],"_adapterSettings":{"displayFormat":"auto","showDisassembly":"auto","dereferencePointers":true,"suppressMissingSourceFiles":true,"evaluationTimeout":5,"consoleMode":"commands","sourceLanguages":null,"terminalPromptClear":null,"evaluateForHovers":true,"commandCompletions":true,"reproducer":false},"__sessionId":"a7952983-e9df-4813-9e46-4cc95feb3b86"},"type":"request","seq":2}
[2022-12-07T14:33:21.157Z DEBUG codelldb::dap_codec] <-- {"seq":2,"type":"event","event":"output","body":{"category":"stderr","output":"Could not initialize Python interpreter - some features will be unavailable (e.g. debug visualizers).\n"}}
[2022-12-07T14:33:21.157Z DEBUG codelldb::dap_codec] <-- {"seq":3,"type":"event","event":"output","body":{"output":"Console is in 'commands' mode, prefix expressions with '?'.\n"}}
[2022-12-07T14:33:21.204Z DEBUG codelldb::dap_codec] <-- {"seq":4,"type":"event","event":"initialized"}
[2022-12-07T14:33:21.204Z DEBUG codelldb::debug_session] Debug event: 0x7fc336f08548 Event: broadcaster = 0x7fc327c9f440 (lldb.target), type = 0x00000002 (modules-loaded), data = {play00}
[2022-12-07T14:33:21.204Z DEBUG codelldb::dap_codec] <-- {"seq":5,"type":"event","event":"module","body":{"module":{"addressRange":"FFFFFFFFFFFFFFFF","id":"FFFFFFFFFFFFFFFF","name":"play00","path":"/Users/anton/Workspace/tutorials/play00/target/debug/play00","symbolFilePath":"/Users/anton/Workspace/tutorials/play00/target/debug/play00","symbolStatus":"Symbols loaded."},"reason":"new"}}
[2022-12-07T14:33:21.204Z DEBUG codelldb::dap_codec] <-- {"seq":6,"type":"request","command":"runInTerminal","arguments":{"args":["/nix/store/7fryvvdiv8m2iqkbjnj8nja579q7fnb7-vscode-extension-vadimcn-vscode-lldb-1.8.1/share/vscode/extensions/vadimcn.vscode-lldb/adapter/.codelldb-wrapped","terminal-agent","--connect=49856"],"cwd":"","kind":"integrated","title":"Debug"}}
[2022-12-07T14:33:21.208Z DEBUG codelldb::dap_codec] --> {"command":"setBreakpoints","arguments":{"source":{"name":"main.rs","path":"/Users/anton/Workspace/tutorials/play00/src/main.rs"},"lines":[8],"breakpoints":[{"line":8}],"sourceModified":false},"type":"request","seq":3}
[2022-12-07T14:33:21.208Z DEBUG codelldb::dap_codec] --> {"command":"setFunctionBreakpoints","arguments":{"breakpoints":[]},"type":"request","seq":4}
[2022-12-07T14:33:21.208Z DEBUG codelldb::dap_codec] --> {"command":"setDataBreakpoints","arguments":{"breakpoints":[]},"type":"request","seq":5}
[2022-12-07T14:33:21.208Z DEBUG codelldb::dap_codec] --> {"command":"setExceptionBreakpoints","arguments":{"filters":["rust_panic"]},"type":"request","seq":6}
[2022-12-07T14:33:21.224Z DEBUG codelldb::debug_session] Debug event: 0x7fc336f1f608 Event: broadcaster = 0x7fc327c9f440 (lldb.target), type = 0x00000001 (breakpoint-changed), data = {}
[2022-12-07T14:33:21.224Z DEBUG codelldb::dap_codec] <-- {"seq":7,"type":"response","request_seq":3,"success":true,"command":"setBreakpoints","body":{"breakpoints":[{"id":1,"line":8,"message":"Resolved locations: 0","verified":true}]}}
[2022-12-07T14:33:21.224Z DEBUG codelldb::dap_codec] <-- {"seq":8,"type":"response","request_seq":4,"success":true,"command":"setFunctionBreakpoints","body":{"breakpoints":[]}}
[2022-12-07T14:33:21.224Z DEBUG codelldb::dap_codec] <-- {"seq":9,"type":"response","request_seq":5,"success":true,"command":"setDataBreakpoints","body":{"breakpoints":[]}}
[2022-12-07T14:33:21.240Z DEBUG codelldb::debug_session] Debug event: 0x7fc34784f428 Event: broadcaster = 0x7fc327c9f440 (lldb.target), type = 0x00000001 (breakpoint-changed), data = {}
[2022-12-07T14:33:21.240Z DEBUG codelldb::dap_codec] <-- {"seq":10,"type":"response","request_seq":6,"success":true,"command":"setExceptionBreakpoints"}
[2022-12-07T14:33:21.350Z DEBUG codelldb::dap_codec] --> {"command":"configurationDone","type":"request","seq":7}
[2022-12-07T14:33:21.350Z DEBUG codelldb::dap_codec] <-- {"seq":11,"type":"response","request_seq":7,"success":true,"command":"configurationDone"}
[2022-12-07T14:33:21.355Z DEBUG codelldb::dap_codec] --> {"command":"threads","type":"request","seq":8}
[2022-12-07T14:33:21.355Z DEBUG codelldb::dap_codec] <-- {"seq":12,"type":"response","request_seq":8,"success":true,"command":"threads","body":{"threads":[]}}
[2022-12-07T14:33:22.672Z DEBUG codelldb::dap_codec] --> {"type":"response","seq":9,"command":"runInTerminal","request_seq":6,"success":true,"body":{"shellProcessId":15045}}
[2022-12-07T14:33:23.246Z DEBUG codelldb::dap_codec] <-- {"seq":13,"type":"event","event":"output","body":{"output":"Launching: /Users/anton/Workspace/tutorials/play00/target/debug/play00\n"}}
[2022-12-07T14:33:23.894Z ERROR codelldb::debug_session] Connection shut down by remote side while waiting for reply to initial handshake packet
[2022-12-07T14:33:23.894Z DEBUG codelldb::dap_codec] <-- {"seq":14,"type":"response","request_seq":2,"success":false,"command":"","message":"Connection shut down by remote side while waiting for reply to initial handshake packet","show_user":true}
[2022-12-07T14:33:23.916Z DEBUG codelldb::dap_codec] --> {"command":"disconnect","arguments":{"restart":false},"type":"request","seq":10}
[2022-12-07T14:33:23.916Z ERROR codelldb::debug_session] Internal debugger error: Sending isconnect packet failed.
[2022-12-07T14:33:23.916Z DEBUG codelldb::dap_codec] <-- {"seq":15,"type":"response","request_seq":10,"success":false,"command":"","message":"Internal debugger error: Sending isconnect packet failed.","show_user":true}
[2022-12-07T14:33:24.403Z DEBUG codelldb::dap_session] Client has disconnected
[2022-12-07T14:33:24.403Z DEBUG codelldb::debug_session] End of the requests stream
[2022-12-07T14:33:24.403Z DEBUG codelldb::debug_session] DebugSession::drop()
[2022-12-07T14:33:24.403Z DEBUG codelldb] End of the debug session
[2022-12-07T14:33:24.414Z DEBUG codelldb] Exiting
Debug adapter exit code=0 (0x0), signal=null.
```

**Comment by nigelgbanks** [comment]
I've unfortunately not had time to dig into this further.

**Comment by superherointj** [comment]
Supposedly fixed by #211321
Re-open in case issue persist.
